### ISSUE TITLE
CursorMoved event closes float to early if there is a pending cursor move

### ISSUE THREAD

**Issue by glepnir** [neovim/neovim#12923]
- `nvim --version`: v0.5.0-f34eeba2d
- `vim -u DEFAULTS` (version: ) behaves differently?  only neovim 
- Operating system/version: macos 10.15.6
- Terminal name/version: kitty
- `$TERM`: xterm-kitty

### Steps to reproduce using `nvim -u NORC`

```lua
function  jump_to()
  api.nvim_win_set_cursor(0,{4,3})
  vim.lsp.util.open_floating_preview({"test"},'markdown')
end
```

### Actual behaviour

 The `nvim_win_set_cursor` doesn't immediately trigger CursorMoved. The util function opens a float and immediately registers the CursorMoved aucmd. On next event loop "tick", the float is closed even if the cursor was not moved again.

### Expected behaviour

The autocmd being debounced/delayed somehow. Like wait for two `CursorMoved` before closing the window (if one is pending already).

**Comment by glepnir** [comment]
@bfredl  Post here.

**Comment by bfredl** [comment]
Nice, I took the liberty to update the issue header with some descriptions.

**Comment by glepnir** [comment]
@bfredl  Of course you can. but I need another way to solve my problem. maybe we should talk in gitter.

**Comment by zeertzjq** [comment]
I don't think this can be fixed without causing more incompatibilities with Vim or existing plugins.

**Comment by guilhas07** [comment]
Noticed the same issue today when trying to execute: 
```lua
vim.diagnostic.goto_next({ float = false })
vim.diagnostic.open_float()
```
Using vim.defer_fn seems to work for my use case:
```lua
vim.diagnostic.goto_next({ float = false })
vim.defer_fn(vim.diagnostic.open_float, 50)
```
